### PR TITLE
Suppression list paging

### PR DESCRIPTION
**Pull request by cohnmarc**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by olydis** [comment]
@cohnmarc Please complete the checklist. Note that there are CI errors that made me wonder: You updated the old Swagger (2017-03-31) to match the new one (2017-04-19) exactly (was that intended? You want to have a breaking change in an existing API version?). However, the examples for 2017-03-31 don't match the changes you did (paging), so our model validation fails.

**Comment by cohnmarc** [comment]
Rolled back version 2017-03-31 swagger output. It was incorrectly commited

File: arm-advisor/2017-04-19/swagger/advisor.json [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+				"description": "Retrieves the list of snoozed or dismissed suppressions for a subscription. The snoozed or dismissed attribute of a recommendation is referred to as a suppression.",
+				"operationId": "Suppressions_List",
+				"parameters": [{
+						"$ref": "#/parameters/subscriptionIdParameter"
+					}, {
+						"$ref": "#/parameters/apiVersionParameter"
+					}, {
+						"name": "$top",
+						"in": "query",
+						"description": "The number of suppressions per page if a paged version of this API is being used.",
```
Comment (vishrutshah) [inline arm-advisor/2017-04-19/swagger/advisor.json]: I believe the `$top` would be `The maximum number of suppression to return from the operation`. Now if your paging mechanism accommodates 50 iteams per page and user has asked for 55 then you do paging and if user has asked for 45 then you do not. Correct me if i am misunderstanding `$top` usage

File: arm-advisor/2017-04-19/swagger/advisor.json [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+						"required": false,
+						"type": "integer"
```
Comment (vishrutshah) [inline arm-advisor/2017-04-19/swagger/advisor.json]: You should also add ` "format": "int32"`. That way typed languages can efficiently use the memory for this variable.

File: arm-advisor/2017-04-19/swagger/advisor.json [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+					}, {
+						"name": "$skipToken",
+						"in": "query",
+						"description": "The page-continuation token to use with a paged version of this API.",
```
Comment (vishrutshah) [inline arm-advisor/2017-04-19/swagger/advisor.json]: I am not sure whether we want to say `with a paged version of this API`. I'd prefer only `The page-continuation token`

**Review by vishrutshah — CHANGES_REQUESTED** [review]
Changes look good. Just left few feedback. 

Also opened a bug in our tool that seems like not interpreting the type in query parameters. Ref: https://github.com/Azure/oav/issues/111

File: arm-advisor/2017-04-19/swagger/advisor.json (hunk 1 above)
Comment (cohnmarc) [inline arm-advisor/2017-04-19/swagger/advisor.json]: Users can use both paged and not paged 2017-04-19. If user do not specify top then all items will be retuned up to limit

Comment (cohnmarc) [inline arm-advisor/2017-04-19/swagger/advisor.json]: top is the requested by user page size per subscription, so users can limit how many items returned

Comment (vishrutshah) [inline arm-advisor/2017-04-19/swagger/advisor.json]: okay then

Comment (vishrutshah) [inline arm-advisor/2017-04-19/swagger/advisor.json]: okay

**Review by vishrutshah — APPROVED** [review]
Changes LGTM! 

Waiting for CI to pass.

File: arm-advisor/2017-04-19/swagger/advisor.json (hunk 3 above)
Comment (cohnmarc) [inline arm-advisor/2017-04-19/swagger/advisor.json]: It is aligned with our recommendation paging message we already give to our customers

**Pull request merged**